pynamodb/models.py

```diff
@@ -447,7 +448,12 @@ def from_raw_data(cls, data):
         for name, value in mutable_data.items():
             attr = cls._get_attributes().get(name, None)
             if attr:
-                kwargs[name] = attr.deserialize(value.get(ATTR_TYPE_MAP[attr.attr_type]))
+                serialized_dynamo_type = ATTR_TYPE_MAP[attr.attr_type]
+                value_to_deserialize = value.get(serialized_dynamo_type)
+                # this only needs to be here until models have all BOOL types serialized
```

Comment:
You should assume that's pretty much until the end of time and just handle it gracefully with a comment about why we need to do this (legacy compat)